openmdao/utils/options_dictionary.py

```diff
@@ -43,6 +49,9 @@ def __init__(self, parent_name=None, read_only=False):
 
         self._all_recordable = True
 
+        self._setitem_warning_issued = []
```

Comment:
Do we really need to bookkeep setitem and getitem warnings separately?

Reply:
I kind of thought so but maybe not. I can't think of a scenario where it would be really critical to do both. I will make it one. Thanks